contracts/ERC20/internal/properties/ERC20BasicProperties.sol

```diff
@@ -289,4 +289,30 @@ abstract contract CryticERC20BasicProperties is CryticERC20Base {
             );
         }
     }
+
+    // TransferFrom for more than allowance should not be allowed
+    function test_ERC20_transferFromMoreThanAllowance(
+        address target
+    ) public {
+        uint256 balance_sender = balanceOf(msg.sender);
+        uint256 balance_receiver = balanceOf(target);
+        uint256 allowance = allowance(msg.sender, address(this));
+        require(balance_sender > 0 && allowance < balance_sender);
+
+        bool r = this.transferFrom(msg.sender, target, allowance + 1);
+        assertWithMsg(
+            r == false,
+            "Successful transferFrom for more than allowance"
+        );
+        assertEq(
+            balanceOf(msg.sender),
+            balance_sender,
+            "TransferFrom for more than amount approved source allowance"
```

Comment:
i'm not sure if i understand the 4 error messages, if given out of context (like, when the property fails).
something like "TransferFrom for more than approval affected sender/receiver balance" should be clearer, what do you think?

Reply:
I updated the error messages, let me know if they sound clearer now.